tests/abc_atlas_cache/test_abc_project_cache.py

```diff
@@ -249,3 +252,116 @@ def test_abc_project_cache_local_cache(self):
+        cache = AbcProjectCache.from_local_cache(self.cache_dir)
+        assert cache.current_manifest == f'releases/{self.new_version}/manifest.json'  # noqa: E501
+        assert cache.list_all_downloaded_manifests == [
+            f'releases/{self.new_version}/manifest.json'
+        ]
+        assert cache.list_manifest_file_names == [
+            f'releases/{self.new_version}/manifest.json'
+        ]
+        assert cache.list_directories == ["second_dir", "test_directory"]
+        assert cache.list_data_files("second_dir") == [self.new_file]
+        assert cache.list_data_files("test_directory") == [self.data_file]
+        assert cache.list_metadata_files("test_directory") == [
+            "metadata_file"
+        ]
+        with pytest.raises(KeyError, match="first_dir"):
+            cache.list_data_files("first_dir")
+
+        data_path = cache.get_data_path(directory="second_dir",
+                                        file_name=self.new_file)
+        assert data_path.exists()
+        assert data_path == self.cache_dir / self.new_file_attr.relative_path
+        data_path = cache.get_data_path(directory="test_directory",
+                                        file_name=self.data_file)
+        assert data_path.exists()
+        assert data_path == self.cache_dir / self.new_data_path
+        metadata_path = cache.get_metadata_path(directory="test_directory",
+                                                file_name="metadata_file")
+        assert metadata_path.exists()
+        assert metadata_path == self.cache_dir / self.new_metadata_path
+
+        data_paths = cache.get_directory_data(directory="test_directory")
+        assert data_paths == [self.cache_dir / self.new_data_path]
+        metadata_paths = cache.get_directory_metadata(
+            directory="test_directory"
+        )
+        assert metadata_paths == [self.cache_dir / self.new_metadata_path]
+
+    def test_abc_project_cache_from_cache_dir(self):
+        """Run a suite of integration tests on the AbcProjectCache class
+        from_local_cache.
+        """
+        AbcProjectCache._bucket_name = self.test_bucket_name
+        cache = AbcProjectCache.from_cache_dir(self.cache_dir)
+
+        # Download data into the local cache.
+        cache.get_data_path(directory="second_dir",
+                            file_name=self.new_file)
+        cache.get_data_path(directory="test_directory",
+                            file_name=self.data_file)
+        cache.get_metadata_path(directory="test_directory",
+                                file_name="metadata_file")
+
+        # Remove the previous cache object.
+        del cache
+
+        # Initialize a new read only object from the local cache.
+        with patch('os.access', return_value=False):
+            cache = AbcProjectCache.from_cache_dir(self.cache_dir)
+        assert cache.current_manifest == f'releases/{self.new_version}/manifest.json'  # noqa: E501
```

Comment:
Should there be a line in this test that verifies `isinstance(cache.cache, LocalCache)` to make sure the new method instantiated a `LocalCache` rather than an `S3Cache`?

Reply:
Sounds good. Would simplify the test significantly as all the other stuff is already tested.